File: 이것이 취업을 위한 코딩 테스트다/이진_탐색/순차_탐색/순차_탐색.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int sequentialSearch(int n, string target, vector<string> arr) {
  for (int i = 0; i < n; i++) {
    if (arr[i] == target) {
      return i + 1;
    }
  }
  return -1;
}

int main() {
  int n;
  string target;
  vector<string> arr;

  cin >> n >> target;

  for (int i = 0; i < n; i++) {
    string x;
    cin >> x;
    arr.push_back(x);
  }

  cout << sequentialSearch(n, target, arr);

  return 0;
}

File: 이것이 취업을 위한 코딩 테스트다/이진_탐색/순차_탐색/순차_탐색.java
import java.util.Scanner;

public class 순차_탐색 {
  public static int sequantialSearch(int n, String target, String[] arr) {
    for (int i = 0; i < n; i++) {
      System.out.println(arr[i]);

      if (arr[i].equals(target)) {
        return i + 1;
      }
    }
    return - 1;
  }
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);

    int n = sc.nextInt();
    String target = sc.next();

    String[] arr = new String[n];
    for (int i = 0; i < n; i++) {
      arr[i] = sc.next();
    }

    System.out.println(sequantialSearch(n, target, arr));
  }
}
